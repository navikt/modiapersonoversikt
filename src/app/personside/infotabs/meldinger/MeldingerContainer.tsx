import * as React from 'react';
import { Traad } from '../../../../models/meldinger/meldinger';
import styled from 'styled-components';
import { pxToRem } from '../../../../styles/personOversiktTheme';
import RestResourceConsumer from '../../../../rest/consumer/RestResourceConsumer';
import TraadListe from './traadliste/TraadListe';
import { CenteredLazySpinner } from '../../../../components/LazySpinner';
import { useEffect, useState } from 'react';
import { hasData } from '../../../../rest/utils/restResource';
import { huskForrigeValgtTraad } from '../../../../redux/meldinger/actions';
import { useDispatch } from 'react-redux';
import { useAppState, useRestResource } from '../../../../utils/customHooks';
import { MeldingerDyplenkeRouteComponentProps, useInfotabsDyplenker, useValgtTraadIUrl } from '../dyplenker';
import { withRouter } from 'react-router';
import TraadVisning from './traadvisning/TraadVisning';
import Verktoylinje from './traadvisning/verktoylinje/Verktoylinje';
import { AlertStripeInfo } from 'nav-frontend-alertstriper';
import { ScrollBar, scrollBarContainerStyle } from '../utils/InfoTabsScrollBar';
import { useSokEtterMeldinger } from './utils/meldingerUtils';

interface TraadVisningWrapperProps {
    valgtTraad?: Traad;
    sokeord: string;
    traaderEtterSok: Traad[];
}
const meldingerMediaTreshold = pxToRem(1000);

const MeldingerArticleStyle = styled.article`
    ${scrollBarContainerStyle(meldingerMediaTreshold)};
    @media (min-width: ${meldingerMediaTreshold}) {
        height: 0; /* IE11 */
        flex-grow: 1; /* IE11 */
        display: flex;
        > *:first-child {
            flex: 35% 1 1;
        }
        > *:last-child {
            flex: 65% 1 1;
        }
    }
    position: relative;
`;

function TraadVisningWrapper(props: TraadVisningWrapperProps) {
    if (props.traaderEtterSok.length === 0) {
        return <AlertStripeInfo>Søket ga ingen treff på meldinger</AlertStripeInfo>;
    }
    return (
        <>
            <Verktoylinje valgtTraad={props.valgtTraad} />
            <TraadVisning sokeord={props.sokeord} valgtTraad={props.valgtTraad} />
        </>
    );
}

function MeldingerContainer(props: MeldingerDyplenkeRouteComponentProps) {
    const dispatch = useDispatch();
    const forrigeValgteTraad = useAppState(state => state.meldinger.forrigeValgteTraad);
    const traaderResource = useRestResource(resources => resources.tråderOgMeldinger);
    const dyplenker = useInfotabsDyplenker();
    const traadIUrl = useValgtTraadIUrl(props);
    const [sokeord, setSokeord] = useState('');
    const traaderEtterSok = useSokEtterMeldinger(hasData(traaderResource) ? traaderResource.data : [], sokeord);

    useEffect(
        function visForsteTreffITraadvisning() {
            if (traadIUrl && traaderEtterSok.length > 0 && !traaderEtterSok.includes(traadIUrl)) {
                props.history.push(dyplenker.meldinger.link(traaderEtterSok[0]));
            }
        },
        [traadIUrl, traaderEtterSok, props.history, dyplenker.meldinger]
    );

    useEffect(() => {
        if (!traadIUrl && hasData(traaderResource)) {
            if (traaderResource.data.length === 0) {
                return;
            }
            props.history.push(dyplenker.meldinger.link(forrigeValgteTraad || traaderResource.data[0]));
        } else if (traadIUrl !== forrigeValgteTraad && !!traadIUrl) {
            dispatch(huskForrigeValgtTraad(traadIUrl));
        }
    }, [forrigeValgteTraad, traadIUrl, traaderResource, dispatch, props.history, dyplenker.meldinger]);

    return (
        <RestResourceConsumer<Traad[]>
            getResource={restResources => restResources.tråderOgMeldinger}
            returnOnPending={<CenteredLazySpinner />}
        >
            {data => {
                if (data.length === 0) {
                    return <AlertStripeInfo>Brukeren har ingen meldinger</AlertStripeInfo>;
                }
                return (
                    <MeldingerArticleStyle>
                        <ScrollBar>
                            <TraadListe
                                sokeord={sokeord}
                                setSokeord={setSokeord}
                                traader={data}
                                valgtTraad={traadIUrl}
                            />
                        </ScrollBar>
                        <ScrollBar>
                            <TraadVisningWrapper
                                traaderEtterSok={traaderEtterSok}
                                sokeord={sokeord}
                                valgtTraad={traadIUrl}
                            />
                        </ScrollBar>
                    </MeldingerArticleStyle>
                );
            }}
        </RestResourceConsumer>
    );
}

export default withRouter(MeldingerContainer);
