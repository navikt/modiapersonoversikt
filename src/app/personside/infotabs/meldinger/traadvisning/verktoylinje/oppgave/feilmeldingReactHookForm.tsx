import { FieldPath, FieldValues } from 'react-hook-form';

export const feilmeldingReactHookForm = <
    TFieldValues extends FieldValues,
    TFieldName extends FieldPath<TFieldValues> = FieldPath<TFieldValues>
>(
    { formState }: FieldValues,
    formKey: TFieldName
) => {
    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
    const { touchedFields, isSubmitted, errors } = formState;

    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access
    const touched = touchedFields[formKey];
    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access
    const error = errors[formKey];

    if ((touched || isSubmitted) && error) {
        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
        return <>{error.message}</>;
    }
    return undefined;
};
