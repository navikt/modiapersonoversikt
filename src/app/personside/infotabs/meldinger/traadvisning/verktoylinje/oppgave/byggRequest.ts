import { OpprettOppgaveRequest } from '../../../../../../../models/meldinger/oppgave';
import { getSaksbehandlerEnhet } from '../../../../../../../utils/loggInfo/saksbehandlersEnhetInfo';
import { eldsteMelding } from '../../../utils/meldingerUtils';
import { InnloggetSaksbehandler } from '../../../../../../../models/innloggetSaksbehandler';
import { OppgaveProps, OppgaveSkjemaProps } from './oppgaveInterfaces';
import { formatterDatoTidNaa } from '../../../../../../../utils/dateUtils';
import { Traad } from '../../../../../../../models/meldinger/meldinger';

export function lagOppgaveRequest(
    props: OppgaveProps,
    form: OppgaveSkjemaProps,
    fodselsnummer: string,
    valgtTraad?: Traad
): OpprettOppgaveRequest {
    const saksbehandlerEnhet = getSaksbehandlerEnhet();
    const valgtTema = form.state.valgtTema;
    const temakode = valgtTema ? valgtTema.kode : 'UKJENT';
    const valgtOppgavetype = form.state.valgtOppgavetype;

    return {
        fnr: fodselsnummer,
        valgtEnhetId: saksbehandlerEnhet ? saksbehandlerEnhet : '2820',
        behandlingskjedeId: valgtTraad ? eldsteMelding(valgtTraad).id : 'UKJENT',
        dagerFrist: valgtOppgavetype ? valgtOppgavetype.dagerFrist : 0,
        ansvarligIdent: form.state.valgtAnsatt!.ident,
        beskrivelse: lagBeskrivelse(form.state.beskrivelse, props.innloggetSaksbehandler, saksbehandlerEnhet),
        temaKode: temakode,
        underkategoriKode: form.state.valgtUnderkategori && form.state.valgtUnderkategori.kode,
        brukerid: props.gjeldendeBrukerFnr,
        oppgaveTypeKode: valgtOppgavetype ? valgtOppgavetype.kode : 'UKJENT',
        prioritetKode: form.state.valgtPrioritet + '_' + temakode,
        ansvarligEnhetId: form.state.valgtEnhet!.enhetId
    };
}

function lagBeskrivelse(
    beskrivelse: string,
    innloggetSaksbehandler: InnloggetSaksbehandler,
    saksbehandlerEnhet?: string
) {
    return `--- ${formatterDatoTidNaa()} ${innloggetSaksbehandler.navn} (${
        innloggetSaksbehandler.ident
    } ${saksbehandlerEnhet}) ---\n${beskrivelse}`;
}
