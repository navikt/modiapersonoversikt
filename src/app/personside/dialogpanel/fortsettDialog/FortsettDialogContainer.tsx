import * as React from 'react';
import { FormEvent, useState } from 'react';
import FortsettDialog from './FortsettDialog';
import { isPosting } from '../../../../rest/utils/postResource';
import { FortsettDialogValidator } from './validatorer';
import { Meldingstype, Traad } from '../../../../models/meldinger/meldinger';
import { setIngenValgtTraadDialogpanel } from '../../../../redux/oppgave/actions';
import { useRestResource } from '../../../../utils/customHooks';
import { useDispatch } from 'react-redux';
import { OppgavelisteValg } from '../sendMelding/SendNyMelding';
import { Oppgave } from '../../../../models/oppgave';
import { JournalforingsSak } from '../../infotabs/meldinger/traadvisning/verktoylinje/journalforing/JournalforingPanel';
import LeggTilbakepanel from './leggTilbakePanel/LeggTilbakepanel';
import { useFortsettDialogKvittering } from './FortsettDialogKvittering';
import useOpprettHenvendelse from './useOpprettHenvendelse';
import { erEldsteMeldingJournalfort } from '../../infotabs/meldinger/utils/meldingerUtils';
import { loggError } from '../../../../utils/frontendLogger';
import { Temagruppe } from '../../../../models/Temagrupper';

export type FortsettDialogType =
    | Meldingstype.SVAR_SKRIFTLIG
    | Meldingstype.DELVIS_SVAR_SKRIFTLIG
    | Meldingstype.SVAR_OPPMOTE
    | Meldingstype.SVAR_TELEFON
    | Meldingstype.SPORSMAL_MODIA_UTGAAENDE;

export interface FortsettDialogState {
    tekst: string;
    dialogType: FortsettDialogType;
    temagruppe?: Temagruppe;
    oppgave?: Oppgave;
    oppgaveListe: OppgavelisteValg;
    sak?: JournalforingsSak;
    visFeilmeldinger: boolean;
}

interface Props {
    traad: Traad;
    tilknyttetOppgave?: Oppgave;
}

function FortsettDialogContainer(props: Props) {
    const initialState = {
        tekst: '',
        dialogType: Meldingstype.SVAR_SKRIFTLIG as FortsettDialogType,
        tema: undefined,
        oppgave: props.tilknyttetOppgave,
        visFeilmeldinger: false,
        sak: undefined,
        oppgaveListe: OppgavelisteValg.MinListe
    };
    const [state, setState] = useState<FortsettDialogState>(initialState);
    const sendSvarResource = useRestResource(resources => resources.sendSvar);
    const sendDelsvarResource = useRestResource(resources => resources.sendDelsvar);
    const reloadMeldinger = useRestResource(resources => resources.tråderOgMeldinger.actions.reload);
    const resetPlukkOppgaveResource = useRestResource(resources => resources.plukkNyeOppgaver.actions.reset);
    const reloadTildelteOppgaver = useRestResource(resources => resources.tildelteOppgaver.actions.reload);
    const leggTilbakeResource = useRestResource(resources => resources.leggTilbakeOppgave);
    const senderMelding =
        isPosting(sendSvarResource) || isPosting(sendDelsvarResource) || isPosting(leggTilbakeResource);
    const dispatch = useDispatch();
    const updateState = (change: Partial<FortsettDialogState>) =>
        setState({
            ...state,
            visFeilmeldinger: false,
            ...change
        });

    const opprettHenvendelse = useOpprettHenvendelse(props.traad);
    const kvittering = useFortsettDialogKvittering();
    if (opprettHenvendelse.success === false) {
        return opprettHenvendelse.placeholder;
    }
    if (kvittering) {
        return kvittering;
    }

    const handleAvbryt = () => dispatch(setIngenValgtTraadDialogpanel());

    const handleSubmit = (event: FormEvent) => {
        event.preventDefault();
        if (senderMelding) {
            return;
        }
        const callback = () => {
            dispatch(resetPlukkOppgaveResource);
            dispatch(reloadTildelteOppgaver);
            dispatch(reloadMeldinger);
        };
        const erOppgaveTilknyttetAnsatt = state.oppgaveListe === OppgavelisteValg.MinListe;
        const oppgaveId = props.tilknyttetOppgave ? props.tilknyttetOppgave.oppgaveid : undefined;
        const commonPayload = {
            fritekst: state.tekst,
            meldingstype: state.dialogType,
            traadId: props.traad.traadId,
            behandlingsId: opprettHenvendelse.behandlingsId,
            oppgaveId: oppgaveId
        };
        if (
            FortsettDialogValidator.erGyldigSvarSkriftlig(state) ||
            FortsettDialogValidator.erGyldigSvarOppmote(state) ||
            FortsettDialogValidator.erGyldigSvarTelefon(state)
        ) {
            dispatch(
                sendSvarResource.actions.post(
                    {
                        ...commonPayload,
                        erOppgaveTilknyttetAnsatt: erOppgaveTilknyttetAnsatt // Hva skal denne være?
                    },
                    callback
                )
            );
        } else if (FortsettDialogValidator.erGyldigSpørsmålSkriftlig(state, props.traad)) {
            const erJournalfort = erEldsteMeldingJournalfort(props.traad);
            if (!state.sak && !erJournalfort) {
                const error = Error('For å opprette spørsmål må meldingen være journalført eller sak må være valgt');
                console.error(error);
                loggError(error);
                return;
            }
            dispatch(
                sendSvarResource.actions.post(
                    {
                        ...commonPayload,
                        erOppgaveTilknyttetAnsatt: erOppgaveTilknyttetAnsatt,
                        saksId: state.sak ? state.sak.saksId : undefined
                    },
                    callback
                )
            );
        } else if (FortsettDialogValidator.erGyldigDelsvar(state) && props.tilknyttetOppgave && state.temagruppe) {
            dispatch(
                sendDelsvarResource.actions.post(
                    {
                        fritekst: state.tekst,
                        traadId: props.traad.traadId,
                        oppgaveId: props.tilknyttetOppgave.oppgaveid,
                        temagruppe: state.temagruppe,
                        behandlingsId: opprettHenvendelse.behandlingsId
                    },
                    callback
                )
            );
        } else {
            updateState({ visFeilmeldinger: true });
        }
    };

    const meldingMedTemagruppe = props.traad.meldinger.find(melding => melding.temagruppe);
    const temagruppe = meldingMedTemagruppe ? meldingMedTemagruppe.temagruppe : Temagruppe.Null;

    return (
        <>
            <FortsettDialog
                handleAvbryt={handleAvbryt}
                state={state}
                updateState={updateState}
                handleSubmit={handleSubmit}
                traad={props.traad}
                key={props.traad.traadId}
                senderMelding={senderMelding}
            />
            {props.tilknyttetOppgave && <LeggTilbakepanel oppgave={props.tilknyttetOppgave} temagruppe={temagruppe} />}
        </>
    );
}

export default FortsettDialogContainer;
