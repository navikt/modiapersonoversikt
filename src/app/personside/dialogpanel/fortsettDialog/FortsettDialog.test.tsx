import * as React from 'react';
import * as renderer from 'react-test-renderer';
import TestProvider from '../../../../test/Testprovider';
import { getTestStore } from '../../../../test/testStore';
import { setValgtTraadDialogpanel } from '../../../../redux/oppgave/actions';
import { statiskTraadMock } from '../../../../mock/meldinger/statiskTraadMock';
import FortsettDialogContainer from './FortsettDialogContainer';

test('viser fortsett dialog', () => {
    const testStore = getTestStore();
    testStore.dispatch(setValgtTraadDialogpanel(statiskTraadMock));
    const dialogPanelBody = renderer.create(
        <TestProvider customStore={testStore}>
            <FortsettDialogContainer traad={statiskTraadMock} />
        </TestProvider>
    );

    expect(dialogPanelBody.toJSON()).toMatchSnapshot();
});
