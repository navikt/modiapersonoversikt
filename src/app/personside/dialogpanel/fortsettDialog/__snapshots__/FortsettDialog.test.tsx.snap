// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`viser fortsett dialog 1`] = `
.c5 {
  width: 100%;
  padding-right: 1rem;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c7 {
  padding: 1rem;
  overflow-wrap: break-word;
}

.c4 {
  border-top: 0.1rem rgba(0,0,0,0.2) solid;
}

.c4.ekspanderbartPanel {
  border-radius: 0;
}

.c8 {
  border-top: 0.1rem rgba(0,0,0,0.2) solid;
}

.c2 .ekspanderbartPanel__innhold {
  padding: 0;
}

.c2 .ekspanderbartPanel__hode:hover {
  box-shadow: inset 0 0 0 0.1rem #0067c5,0 0.1rem 0.1rem rgba(0,0,0,0.2);
}

.c2 > *:not(:first-child) {
  margin-top: 1rem;
}

.c2 > * {
  border: solid 0.0625rem #78706a;
}

.c3 {
  overflow: hidden;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-top: 1rem;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.c1 .skjemaelement {
  margin-bottom: 0;
}

.c1 > * {
  margin-bottom: 1rem;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c13 > * {
  margin-top: 0.3rem;
}

.c13 > *:not(:last-child) {
  margin-right: 1rem;
}

.c10 {
  position: absolute;
  height: 2rem;
  width: 2rem;
  border-bottom-right-radius: 50%;
  padding: 0;
  background-color: #0067c5;
  border: none;
  z-index: 10;
  -webkit-transform: none;
  -ms-transform: none;
  transform: none;
}

.c10:after {
  background: none;
}

.c10:hover {
  -webkit-transform: none;
  -ms-transform: none;
  transform: none;
}

.c11 {
  position: relative;
  top: 2px;
  width: 18px;
}

.c9 {
  position: relative;
}

.c9 textarea {
  min-height: 9rem;
}

.c9 label {
  position: absolute !important;
  height: 1px;
  width: 1px;
  overflow: hidden;
  -webkit-clip: rect(1px 1px 1px 1px);
  clip: rect(1px 1px 1px 1px);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
}

.c12 textarea,
.c12 .textareamirror {
  padding-left: 2.25rem;
}

.c16 label {
  position: absolute !important;
  height: 1px;
  width: 1px;
  overflow: hidden;
  -webkit-clip: rect(1px 1px 1px 1px);
  clip: rect(1px 1px 1px 1px);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
}

.c15 {
  margin-top: 1rem;
}

.c15 label {
  position: absolute !important;
  height: 1px;
  width: 1px;
  overflow: hidden;
  -webkit-clip: rect(1px 1px 1px 1px);
  clip: rect(1px 1px 1px 1px);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
}

.c14 > *:not(:first-child) {
  margin-top: 1rem;
}

.c0 {
  padding: 1rem .8rem;
}

.c17 {
  margin-top: 1rem;
}

.c18 {
  width: 100%;
}

<article
  className="c0"
>
  <h2
    className="typo-undertittel"
  >
    Fortsett dialog
  </h2>
  <form
    className="c1"
    onSubmit={[Function]}
  >
    <div
      className="c2"
    >
      <div
        className="ekspanderbartPanel c3 ekspanderbartPanel--apen"
      >
        <button
          aria-controls="Helt tilfeldig ID"
          aria-expanded={true}
          className="ekspanderbartPanel__hode"
          onClick={[Function]}
          onKeyDown={[Function]}
          type="button"
        >
          <div
            className="ekspanderbartPanel__flex-wrapper"
          >
            <p
              className="typo-ingress"
            >
              Vis tidligere meldinger
            </p>
            <span
              className="ekspanderbartPanel__indikator"
            />
          </div>
        </button>
        <article
          className="ekspanderbartPanel__innhold"
        >
          <div
            className="ekspanderbartPanel c4 ekspanderbartPanel--lukket"
            erEnkeltstaende={false}
          >
            <button
              aria-expanded={false}
              className="ekspanderbartPanel__hode"
              onClick={[Function]}
              onKeyDown={[Function]}
              type="button"
            >
              <div
                className="ekspanderbartPanel__flex-wrapper"
              >
                <div
                  className="c5"
                >
                  <p
                    className="typo-undertekst"
                  >
                    07. juli 2019 00:00
                  </p>
                  <p
                    className="typo-undertekst"
                  >
                    Skrevet av 
                    Sander Hansen (nzxo4q)
                  </p>
                  <div
                    className="c6"
                  >
                    <p
                      className="typo-ingress"
                    >
                      Samtalereferat oppmøte
                    </p>
                    <p
                      className="typo-undertekst-bold"
                    >
                      Arbeid
                    </p>
                  </div>
                </div>
                <span
                  className="ekspanderbartPanel__indikator"
                />
              </div>
            </button>
            <article
              className="ekspanderbartPanel__innhold"
            >
              <div
                className="tekstomrade c7"
              >
                Ea laborum accusamus et ex voluptates provident eveniet voluptatem. Sunt et sed eligendi harum exercitationem. Eaque est maiores autem sunt occaecati atque ipsa ea itaque. Dolorum enim quaerat alias ut sit labore pariatur.
              </div>
            </article>
          </div>
          <div
            className="ekspanderbartPanel c4 ekspanderbartPanel--lukket"
            erEnkeltstaende={false}
          >
            <button
              aria-expanded={false}
              className="ekspanderbartPanel__hode"
              onClick={[Function]}
              onKeyDown={[Function]}
              type="button"
            >
              <div
                className="ekspanderbartPanel__flex-wrapper"
              >
                <div
                  className="c5"
                >
                  <p
                    className="typo-undertekst"
                  >
                    16. juli 2019 00:00
                  </p>
                  <div
                    className="c6"
                  >
                    <p
                      className="typo-ingress"
                    >
                      Spørsmål fra bruker
                    </p>
                    <p
                      className="typo-undertekst-bold"
                    >
                      Arbeid
                    </p>
                  </div>
                </div>
                <span
                  className="ekspanderbartPanel__indikator"
                />
              </div>
            </button>
            <article
              className="ekspanderbartPanel__innhold"
            >
              <div
                className="tekstomrade c7"
              >
                Quo magni eaque et libero minima ipsum amet et dolorem. Sed iusto ratione sit. Voluptate alias qui impedit ut dolorum doloribus qui. Qui beatae rerum minima suscipit.
              </div>
            </article>
          </div>
          <div
            className="ekspanderbartPanel c4 ekspanderbartPanel--lukket"
            erEnkeltstaende={false}
          >
            <button
              aria-expanded={false}
              className="ekspanderbartPanel__hode"
              onClick={[Function]}
              onKeyDown={[Function]}
              type="button"
            >
              <div
                className="ekspanderbartPanel__flex-wrapper"
              >
                <div
                  className="c5"
                >
                  <p
                    className="typo-undertekst"
                  >
                    12. juli 2019 00:00
                  </p>
                  <div
                    className="c6"
                  >
                    <p
                      className="typo-ingress"
                    >
                      Spørsmål fra bruker
                    </p>
                    <p
                      className="typo-undertekst-bold"
                    >
                      Arbeid
                    </p>
                  </div>
                </div>
                <span
                  className="ekspanderbartPanel__indikator"
                />
              </div>
            </button>
            <article
              className="ekspanderbartPanel__innhold"
            >
              <div
                className="tekstomrade c7"
              >
                Animi qui molestiae omnis reiciendis quam in. Provident magnam dolor. Omnis voluptatem placeat labore in ut aliquam perspiciatis fugiat. Perspiciatis voluptatum odit.
              </div>
            </article>
          </div>
        </article>
      </div>
      <div
        className="ekspanderbartPanel c8 ekspanderbartPanel--apen"
        erEnkeltstaende={true}
      >
        <button
          aria-controls="Helt tilfeldig ID"
          aria-expanded={true}
          className="ekspanderbartPanel__hode"
          onClick={[Function]}
          onKeyDown={[Function]}
          type="button"
        >
          <div
            className="ekspanderbartPanel__flex-wrapper"
          >
            <div
              className="c5"
            >
              <p
                className="typo-undertekst"
              >
                27. juli 2019 00:00
              </p>
              <p
                className="typo-undertekst"
              >
                Skrevet av 
                Kristoffer Fredriksen (xy70r8)
              </p>
              <div
                className="c6"
              >
                <p
                  className="typo-ingress"
                >
                  Delsvar
                </p>
                <p
                  className="typo-undertekst-bold"
                >
                  Arbeid
                </p>
              </div>
            </div>
            <span
              className="ekspanderbartPanel__indikator"
            />
          </div>
        </button>
        <article
          className="ekspanderbartPanel__innhold"
        >
          <div
            className="tekstomrade c7"
          >
            Dolores omnis ea voluptatem corrupti odit incidunt enim. Unde fugiat sed animi explicabo maiores. Nihil cum accusamus omnis officiis deserunt est. Qui quisquam unde aut magni dignissimos.
          </div>
        </article>
      </div>
    </div>
    <div
      className="c9"
    >
      <button
        className="knapp c10 knapp--hoved"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <svg
          className="c11"
          height="24px"
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 24 24"
          width="24px"
          x="0px"
          xmlns="http://www.w3.org/2000/svg"
          y="0px"
        >
          <path
            d="M20.854,6.146l-6-6C14.76,0.053,14.632,0,14.5,0h-11C3.224,0,3,0.224,3,0.5v23C3,23.776,3.224,24,3.5,24h17  c0.276,0,0.5-0.224,0.5-0.5v-17C21,6.367,20.947,6.24,20.854,6.146z M7.5,7H12c0.276,0,0.5,0.224,0.5,0.5S12.276,8,12,8H7.5  C7.224,8,7,7.776,7,7.5S7.224,7,7.5,7z M16.5,20h-9C7.224,20,7,19.776,7,19.5S7.224,19,7.5,19h9c0.276,0,0.5,0.224,0.5,0.5  S16.776,20,16.5,20z M16.5,17h-9C7.224,17,7,16.776,7,16.5S7.224,16,7.5,16h9c0.276,0,0.5,0.224,0.5,0.5S16.776,17,16.5,17z   M16.5,14h-9C7.224,14,7,13.776,7,13.5S7.224,13,7.5,13h9c0.276,0,0.5,0.224,0.5,0.5S16.776,14,16.5,14z M16.5,11h-9  C7.224,11,7,10.776,7,10.5S7.224,10,7.5,10h9c0.276,0,0.5,0.224,0.5,0.5S16.776,11,16.5,11z M14.5,6.5v-6l6,6H14.5z"
            fill="currentColor"
          />
        </svg>
        <span
          className="sr-only"
        >
          Standardtekster
        </span>
      </button>
      <div
        className="c12"
      >
        <div
          className="skjemaelement textarea__container"
        >
          <label
            className="skjemaelement__label"
            htmlFor="Helt tilfeldig ID"
          >
            Melding
          </label>
          <div
            className="textarea--medMeta__wrapper"
          >
            <textarea
              className="skjemaelement__input textarea--medMeta"
              id="Helt tilfeldig ID"
              onChange={[Function]}
              placeholder="Alt du skriver i denne boksen blir synlig for Aremark når du trykker \\"Del med Aremark\\""
              style={
                Object {
                  "height": "30px",
                }
              }
              type="text"
              value=""
            />
            <p
              className="textarea--medMeta__teller"
            >
              <span
                aria-live="polite"
                className="teller-tekst"
              >
                Du har 5000 tegn igjen
              </span>
            </p>
          </div>
          <div
            aria-live="assertive"
            role="alert"
          />
          <div
            aria-hidden="true"
            className="textareamirror"
          />
        </div>
      </div>
      <div
        aria-live="assertive"
        role="alert"
      />
    </div>
    <div
      className="c13"
    >
      <div
        className="skjemaelement skjemaelement--horisontal"
      >
        <input
          checked={true}
          className="skjemaelement__input radioknapp"
          id="Helt tilfeldig ID"
          name="dialogtype"
          onChange={[Function]}
          type="radio"
        />
        <label
          className="skjemaelement__label"
          htmlFor="Helt tilfeldig ID"
        >
          Svar
        </label>
      </div>
      <div
        className="skjemaelement skjemaelement--horisontal"
      >
        <input
          checked={false}
          className="skjemaelement__input radioknapp"
          id="Helt tilfeldig ID"
          name="dialogtype"
          onChange={[Function]}
          type="radio"
        />
        <label
          className="skjemaelement__label"
          htmlFor="Helt tilfeldig ID"
        >
          Delvis svar
        </label>
      </div>
    </div>
    <div
      className=""
    >
      <div
        className="alertstripe alertstripe--info"
      >
        <span
          aria-label="info"
          className="alertstripe__ikon"
        >
          <svg
            focusable="false"
            height="1.5em"
            kind="info-sirkel-fyll"
            viewBox="0 0 24 24"
            width="1.5em"
          >
            <g
              fill="none"
            >
              <path
                d="M12 0C5.382 0 0 5.382 0 12s5.382 12 12 12c6.617 0 12-5.382 12-12S18.617 0 12 0z"
                fill="#669DB4"
              />
              <path
                d="M12 5a1.566 1.566 0 1 1 .11 3.13A1.566 1.566 0 0 1 12 5zm2.976 12.01c.563 0 1.043.431 1.043.991s-.48.992-1.043.992H9.39c-.564 0-1.043-.431-1.043-.992 0-.56.479-.99 1.043-.99h1.6v-5.016h-.986c-.565 0-1.044-.43-1.044-.991 0-.56.48-.991 1.044-.991h2.03c.563 0 1.043.43 1.043.99v6.007h1.899z"
                fill="#FFF"
              />
            </g>
          </svg>
        </span>
        <div
          className="typo-normal alertstripe__tekst"
        >
          Bruker kan ikke svare
        </div>
      </div>
      <div
        className="c14"
      >
        <div
          className="alertstripe alertstripe--info"
        >
          <span
            aria-label="info"
            className="alertstripe__ikon"
          >
            <svg
              focusable="false"
              height="1.5em"
              kind="info-sirkel-fyll"
              viewBox="0 0 24 24"
              width="1.5em"
            >
              <g
                fill="none"
              >
                <path
                  d="M12 0C5.382 0 0 5.382 0 12s5.382 12 12 12c6.617 0 12-5.382 12-12S18.617 0 12 0z"
                  fill="#669DB4"
                />
                <path
                  d="M12 5a1.566 1.566 0 1 1 .11 3.13A1.566 1.566 0 0 1 12 5zm2.976 12.01c.563 0 1.043.431 1.043.991s-.48.992-1.043.992H9.39c-.564 0-1.043-.431-1.043-.992 0-.56.479-.99 1.043-.99h1.6v-5.016h-.986c-.565 0-1.044-.43-1.044-.991 0-.56.48-.991 1.044-.991h2.03c.563 0 1.043.43 1.043.99v6.007h1.899z"
                  fill="#FFF"
                />
              </g>
            </svg>
          </span>
          <div
            className="typo-normal alertstripe__tekst"
          >
            Bruker kan svare
          </div>
        </div>
        <div
          className="skjemaelement c15"
        >
          <label
            className="skjemaelement__label"
            htmlFor="Helt tilfeldig ID"
          >
            Oppgaveliste
          </label>
          <div
            className="selectContainer input--fullbredde"
          >
            <select
              className="skjemaelement__input"
              disabled={false}
              id="Helt tilfeldig ID"
              onChange={[Function]}
              value="MinListe"
            >
              <option
                value="MinListe"
              >
                Svar skal til min oppgaveliste
              </option>
              <option
                value="Enhetensliste"
              >
                Svar skal til 
                 sin oppgaveliste
              </option>
            </select>
          </div>
          <div
            aria-live="assertive"
            role="alert"
          />
        </div>
      </div>
      <div
        className=""
      >
        <div
          className="skjemaelement c16"
        >
          <label
            className="skjemaelement__label"
            htmlFor="Helt tilfeldig ID"
          >
            Tema
          </label>
          <div
            className="selectContainer input--fullbredde"
          >
            <select
              className="skjemaelement__input"
              disabled={false}
              id="Helt tilfeldig ID"
              onChange={[Function]}
              value=""
            >
              <option
                disabled={true}
                value=""
              >
                Velg temagruppe
              </option>
              <option
                value="FMLI"
              >
                Familie
              </option>
              <option
                value="HJLPM"
              >
                Hjelpemidler
              </option>
              <option
                value="BIL"
              >
                Hjelpemidler bil
              </option>
              <option
                value="ORT_HJE"
              >
                Ortopediske hjelpemidler
              </option>
              <option
                value="PENS"
              >
                Pensjon
              </option>
              <option
                value="PLEIEPENGERSY"
              >
                Pleiepenger sykt barn
              </option>
              <option
                value="UFRT"
              >
                Uføretrygd
              </option>
              <option
                value="UTLAND"
              >
                Utland
              </option>
            </select>
          </div>
          <div
            aria-live="assertive"
            role="alert"
          />
        </div>
        <div
          aria-live="assertive"
          role="alert"
        />
      </div>
    </div>
    <button
      className="knapp c17 knapp--hoved"
      disabled={false}
      type="submit"
    >
      Del med Aremark
    </button>
    <button
      className="knapp c18 knapp--flat"
      disabled={false}
      onClick={[Function]}
      type="reset"
    >
      Avbryt
    </button>
  </form>
</article>
`;
